### PR TITLE
handle not found error in purge-chain

### LINKED ISSUE #2273: purge-chain will fail if no old data to delete
```
$ cargo run -- purge-chain --dev
    Finished dev [unoptimized + debuginfo] target(s) in 0.75s
     Running `target/debug/substrate-kitties purge-chain --dev`
Are you sure to remove "/Users/bryan/Library/Application Support/substrate-kitties/chains/dev/db"? (y/n)y
Error: No such file or directory (os error 2)
```

This is project generated from node-template with script from substrate-up.

### PR DESCRIPTION
**Pull request by xlc**
Fixes #2273

### CODE REVIEW

File: core/cli/src/lib.rs [hunk 1]
```diff
@@ -673,8 +673,15 @@ where
 		}
 	}
 
-	fs::remove_dir_all(&db_path)?;
-	println!("{:?} removed.", &db_path);
+	if let Result::Err(err) = fs::remove_dir_all(&db_path) {
```
Comment (bkchr) [inline core/cli/src/lib.rs]: I would like to have a match here and also please move the `Ok(())` into this match statement.

File: core/cli/src/lib.rs [hunk 2]
```diff
@@ -673,10 +673,17 @@ where
...
+	match fs::remove_dir_all(&db_path) {
+		Result::Ok(_) => {
+			println!("{:?} removed.", &db_path);
+			Ok(())
+		},
+		Result::Err(ref err) if err.kind() == ErrorKind::NotFound => {
+			println!("{:?} did not exist.", &db_path);
+			Ok(())
+		},
+		Result::Err(err) => Result::Err(err.into())
```
Comment (xlc) [inline core/cli/src/lib.rs]: not sure if there is more idiomatic way to write this

**Comment by xlc** [comment]
Do you want to back port this to v1.0?

**Comment by bkchr** [comment]
Backport: https://github.com/paritytech/substrate/pull/2279

**Pull request merged**